pkg/instrumentation/sdk.go

```diff
@@ -64,35 +64,54 @@ func (i *sdkInjector) inject(ctx context.Context, insts languageInstrumentations
 		}
 	}
 
-	// inject only to the first container for now
-	// in the future we can define an annotation to configure this
 	if insts.Java != nil {
 		otelinst := *insts.Java
-		i.logger.V(1).Info("injecting java instrumentation into pod", "otelinst-namespace", otelinst.Namespace, "otelinst-name", otelinst.Name)
-		pod = injectJavaagent(i.logger, otelinst.Spec.Java, pod, index)
+		var err error
+		i.logger.V(1).Info("injecting Java instrumentation into pod", "otelinst-namespace", otelinst.Namespace, "otelinst-name", otelinst.Name)
+		pod, err = injectJavaagent(otelinst.Spec.Java, pod, index)
+		if err != nil {
+			i.logger.Info("Skipping javaagent injection", "reason", err.Error(), "container", pod.Spec.Containers[index].Name)
+			return pod
```

Comment:
As per my understanding, [inject](https://github.com/avadhut123pisal/opentelemetry-operator/blob/2a6569c3b168e9f80dda10916d96579bcf4993d5/pkg/instrumentation/sdk.go#L53) function gets called for a single container at a time. So, there should be only one language instrumentation is required for that particular container.  

@pellared  Please let me know, if I'm missing something.

Reply:
> there should be only one language instrumentation is required for that particular container

I do not see any docs, code, nor reason that would disallow injecting more language instrumentations. You can have a container that has more than one process.

Reply:
Agreed.

Reply:
@pavolloffay Looking at the current implementation, it seems that using multiple instrumentations for the same container will not work. One reason is the duplicate volume mounts, because we use the same mount path. There might be some others things also that can break in context of init container. 

I tried adding the annotations for two different language instrumentations, it failed with the error;
` Error creating: Pod "spring-petclinic-5d6d58d9b8-pp268" is invalid: spec.containers[0].volumeMounts[2].mountPath: Invalid value: "/otel-auto-instrumentation": must be unique`

Reply:
The goal we had was to support multiple instrumentations for a single pod (e.g. one container java and other python etc.) 

@avadhut123pisal feel free to book an issue to resolve this limitation if it is important or open a PR to document this in the readme.

Reply:
@pellared  Considering the current implementation (multiple instrumentations for a single pod) I don't think `return` statement would cause any issue.

Reply:
> The goal we had was to support multiple instrumentations for a single pod

@avadhut123pisal I suggest doing the following:

1. Change the `inject` function implementation in a way as if support for multiple instrumentations for a single pod is working (e.g. by using `else` instead of `return pod` in case of an error).
2. Create an issue.
3. Document it in `README.md` as a known issue.

@pavolloffay Does it seem reasonable?

Reply:
> The goal we had was to support multiple instrumentations for a single pod (e.g. one container java and other python etc.)
> 
> @avadhut123pisal feel free to book an issue to resolve this limitation if it is important or open a PR to document this in the readme.

sure !

Reply:
> > @pellared Considering the current implementation (multiple instrumentations for a single pod) I don't think return statement would cause any issue.
> 
> The code is written in an unmaintainable way. The "error handling" suggests that it works only for one instrumentation. The "happy path scenario" suggests that it supports multiple instrumentations.

Yeah. I got your point :)